OpenRA.Game/Map/MapCache.cs

```diff
@@ -176,34 +175,31 @@ void LoadAsyncInternal()
 						bitmap = Minimap.RenderMapPreview(modData.DefaultRules.TileSets[p.Map.Tileset], p.Map, modData.DefaultRules, true);
 					}
 
-					// Note: this is not generally thread-safe, but it works here because:
-					//   (a) This worker is the only thread writing to this sheet
-					//   (b) The main thread is the only thread reading this sheet
-					//   (c) The sheet is marked dirty after the write is completed,
-					//       which causes the main thread to copy this to the texture during
-					//       the next render cycle.
-					//   (d) Any partially written bytes from the next minimap is in an
-					//       unallocated area, and will be committed in the next cycle.
-					try
+					Game.RunAfterTick(() =>
 					{
-						p.SetMinimap(sheetBuilder.Add(bitmap));
-					}
-					finally
-					{
-						if (createdPreview)
-							bitmap.Dispose();
-					}
+						try
+						{
+							p.SetMinimap(sheetBuilder.Add(bitmap));
```

Comment:
The map chooser doesn't feel any more janky that usual with this PR (at least to me), so I'm not even sure it's actually an issue in practise.

I'd really rather avoid complicating the code with more threading nastiness unless it's a very clear win.